src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -57,19 +59,19 @@ public WinInetProxyHelper()
             }
         }
 
-        public string AutoConfigUrl { get; }
+        public string AutoConfigUrl => _autoConfigUrl;
 
-        public bool AutoDetect { get; }
+        public bool AutoDetect => _autoDetect;
 
         public bool AutoSettingsUsed => AutoDetect || !string.IsNullOrEmpty(AutoConfigUrl);
 
         public bool ManualSettingsUsed => !string.IsNullOrEmpty(Proxy);
 
         public bool ManualSettingsOnly => !AutoSettingsUsed && ManualSettingsUsed;
 
-        public string Proxy { get; }
+        public string Proxy => _proxy;
 
-        public string ProxyBypass { get; }
+        public string ProxyBypass => _proxyBypass;
```

Comment:
I'm curious why you changed these from auto-props to having manual backing fields?

Reply:
I was wondering that too.  My guess was that having explicit backing fields made it easier to use reflection for the new tests that were written in the PR? But reflection would work against properties as well as fields. So, this wouldn't be strictly necessary.

Reply:
I did this because read-only props can't be set via reflection, and there is no good way to find the backing field for an auto-prop.